File: ts-sdk/src/sdk/models/shared/retentionpolicyassignment.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { UserMini } from "./usermini";
import { RetentionPolicyMini } from "./retentionpolicymini";


export enum RetentionPolicyAssignmentAssignedToTypeEnum {
    Folder = "folder",
    Enterprise = "enterprise",
    MetadataTemplate = "metadata_template"
}


// RetentionPolicyAssignmentAssignedTo
/** 
 * The `type` and `id` of the content that is under
 * retention. The `type` can either be `folder`
 * `enterprise`, or `metadata_template`.
**/
export class RetentionPolicyAssignmentAssignedTo extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: RetentionPolicyAssignmentAssignedToTypeEnum;
}


export class RetentionPolicyAssignmentFilterFields extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=field" })
  field?: string;

  @SpeakeasyMetadata({ data: "json, name=value" })
  value?: string;
}

export enum RetentionPolicyAssignmentTypeEnum {
    RetentionPolicyAssignment = "retention_policy_assignment"
}


// RetentionPolicyAssignment
/** 
 * A retention assignment represents a rule specifying
 * the files a retention policy retains.
 * Assignments can retain files based on their folder or metadata,
 * or hold all files in the enterprise.
**/
export class RetentionPolicyAssignment extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=assigned_at" })
  assignedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=assigned_by" })
  assignedBy?: UserMini;

  @SpeakeasyMetadata({ data: "json, name=assigned_to" })
  assignedTo?: RetentionPolicyAssignmentAssignedTo;

  @SpeakeasyMetadata({ data: "json, name=filter_fields", elemType: RetentionPolicyAssignmentFilterFields })
  filterFields?: RetentionPolicyAssignmentFilterFields[];

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=retention_policy" })
  retentionPolicy?: RetentionPolicyMini;

  @SpeakeasyMetadata({ data: "json, name=start_date_field" })
  startDateField?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: RetentionPolicyAssignmentTypeEnum;
}

File: python-sdk/src/boxsdk/models/shared/retentionpolicyassignment.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import user_mini as shared_user_mini
from ..shared import retentionpolicy_mini as shared_retentionpolicy_mini

class RetentionPolicyAssignmentAssignedToTypeEnum(str, Enum):
    FOLDER = "folder"
    ENTERPRISE = "enterprise"
    METADATA_TEMPLATE = "metadata_template"


@dataclass_json
@dataclasses.dataclass
class RetentionPolicyAssignmentAssignedTo:
    r"""RetentionPolicyAssignmentAssignedTo
    The `type` and `id` of the content that is under
    retention. The `type` can either be `folder`
    `enterprise`, or `metadata_template`.
    """
    
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: Optional[RetentionPolicyAssignmentAssignedToTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

@dataclass_json
@dataclasses.dataclass
class RetentionPolicyAssignmentFilterFields:
    field: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('field') }})
    value: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('value') }})
    
class RetentionPolicyAssignmentTypeEnum(str, Enum):
    RETENTION_POLICY_ASSIGNMENT = "retention_policy_assignment"


@dataclass_json
@dataclasses.dataclass
class RetentionPolicyAssignment:
    r"""RetentionPolicyAssignment
    A retention assignment represents a rule specifying
    the files a retention policy retains.
    Assignments can retain files based on their folder or metadata,
    or hold all files in the enterprise.
    """
    
    assigned_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('assigned_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    assigned_by: Optional[shared_user_mini.UserMini] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('assigned_by') }})
    assigned_to: Optional[RetentionPolicyAssignmentAssignedTo] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('assigned_to') }})
    filter_fields: Optional[list[RetentionPolicyAssignmentFilterFields]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('filter_fields') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    retention_policy: Optional[shared_retentionpolicy_mini.RetentionPolicyMini] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('retention_policy') }})
    start_date_field: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('start_date_field') }})
    type: Optional[RetentionPolicyAssignmentTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
